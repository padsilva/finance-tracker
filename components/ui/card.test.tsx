import { render } from "@testing-library/react";

import {
  Card,
  CardHeader,
  CardFooter,
  CardTitle,
  CardDescription,
  CardContent,
} from "./card";

describe("Card Components", () => {
  it("should render basic card correctly", () => {
    const { container } = render(
      <Card>
        <CardContent>Basic Card Content</CardContent>
      </Card>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render full card with all subcomponents", () => {
    const { container } = render(
      <Card>
        <CardHeader>
          <CardTitle>Card Title</CardTitle>
          <CardDescription>Card Description</CardDescription>
        </CardHeader>
        <CardContent>
          <p>Main content of the card</p>
        </CardContent>
        <CardFooter>
          <p>Card Footer</p>
        </CardFooter>
      </Card>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with custom classNames", () => {
    const { container } = render(
      <Card className="custom-card">
        <CardHeader className="custom-header">
          <CardTitle className="custom-title">Custom Title</CardTitle>
          <CardDescription className="custom-description">
            Custom Description
          </CardDescription>
        </CardHeader>
        <CardContent className="custom-content">
          <p>Custom Content</p>
        </CardContent>
        <CardFooter className="custom-footer">
          <p>Custom Footer</p>
        </CardFooter>
      </Card>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render card with nested content", () => {
    const { container } = render(
      <Card>
        <CardHeader>
          <CardTitle>Nested Content Card</CardTitle>
        </CardHeader>
        <CardContent>
          <Card>
            <CardHeader>
              <CardTitle>Nested Card</CardTitle>
            </CardHeader>
            <CardContent>Nested content</CardContent>
          </Card>
        </CardContent>
      </Card>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with data-testid attributes", () => {
    const { container } = render(
      <Card>
        <CardHeader data-testid="card-header">
          <CardTitle data-testid="card-title">Test Title</CardTitle>
          <CardDescription data-testid="card-description">
            Test Description
          </CardDescription>
        </CardHeader>
        <CardContent data-testid="card-content">
          <p>Test content</p>
        </CardContent>
        <CardFooter data-testid="card-footer">
          <p>Test footer</p>
        </CardFooter>
      </Card>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render card with different content types", () => {
    const { container } = render(
      <Card>
        <CardHeader>
          <CardTitle>Mixed Content</CardTitle>
        </CardHeader>
        <CardContent>
          <div className="space-y-4">
            <p>Text content</p>
            <button type="button">A button</button>
            <div className="grid grid-cols-2 gap-2">
              <div>Grid item 1</div>
              <div>Grid item 2</div>
            </div>
          </div>
        </CardContent>
      </Card>,
    );

    expect(container).toMatchSnapshot();
  });
});
