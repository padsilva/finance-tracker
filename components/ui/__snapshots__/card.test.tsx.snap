// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Components should render basic card correctly 1`] = `
<div>
  <div
    class="rounded-xl border bg-card text-card-foreground shadow"
    data-testid="card-main"
  >
    <div
      class="p-6 pt-0"
      data-testid="card-content"
    >
      Basic Card Content
    </div>
  </div>
</div>
`;

exports[`Card Components should render card with different content types 1`] = `
<div>
  <div
    class="rounded-xl border bg-card text-card-foreground shadow"
    data-testid="card-main"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
      data-testid="card-header"
    >
      <div
        class="font-semibold leading-none tracking-tight"
        data-testid="card-title"
      >
        Mixed Content
      </div>
    </div>
    <div
      class="p-6 pt-0"
      data-testid="card-content"
    >
      <div
        class="space-y-4"
      >
        <p>
          Text content
        </p>
        <button
          type="button"
        >
          A button
        </button>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div>
            Grid item 1
          </div>
          <div>
            Grid item 2
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Components should render card with nested content 1`] = `
<div>
  <div
    class="rounded-xl border bg-card text-card-foreground shadow"
    data-testid="card-main"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
      data-testid="card-header"
    >
      <div
        class="font-semibold leading-none tracking-tight"
        data-testid="card-title"
      >
        Nested Content Card
      </div>
    </div>
    <div
      class="p-6 pt-0"
      data-testid="card-content"
    >
      <div
        class="rounded-xl border bg-card text-card-foreground shadow"
        data-testid="card-main"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
          data-testid="card-header"
        >
          <div
            class="font-semibold leading-none tracking-tight"
            data-testid="card-title"
          >
            Nested Card
          </div>
        </div>
        <div
          class="p-6 pt-0"
          data-testid="card-content"
        >
          Nested content
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Components should render full card with all subcomponents 1`] = `
<div>
  <div
    class="rounded-xl border bg-card text-card-foreground shadow"
    data-testid="card-main"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
      data-testid="card-header"
    >
      <div
        class="font-semibold leading-none tracking-tight"
        data-testid="card-title"
      >
        Card Title
      </div>
      <div
        class="text-sm text-muted-foreground"
      >
        Card Description
      </div>
    </div>
    <div
      class="p-6 pt-0"
      data-testid="card-content"
    >
      <p>
        Main content of the card
      </p>
    </div>
    <div
      class="flex items-center p-6 pt-0"
      data-testid="card-footer"
    >
      <p>
        Card Footer
      </p>
    </div>
  </div>
</div>
`;

exports[`Card Components should render with custom classNames 1`] = `
<div>
  <div
    class="rounded-xl border bg-card text-card-foreground shadow custom-card"
    data-testid="card-main"
  >
    <div
      class="flex flex-col space-y-1.5 p-6 custom-header"
      data-testid="card-header"
    >
      <div
        class="font-semibold leading-none tracking-tight custom-title"
        data-testid="card-title"
      >
        Custom Title
      </div>
      <div
        class="text-sm text-muted-foreground custom-description"
      >
        Custom Description
      </div>
    </div>
    <div
      class="p-6 pt-0 custom-content"
      data-testid="card-content"
    >
      <p>
        Custom Content
      </p>
    </div>
    <div
      class="flex items-center p-6 pt-0 custom-footer"
      data-testid="card-footer"
    >
      <p>
        Custom Footer
      </p>
    </div>
  </div>
</div>
`;

exports[`Card Components should render with data-testid attributes 1`] = `
<div>
  <div
    class="rounded-xl border bg-card text-card-foreground shadow"
    data-testid="card-main"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
      data-testid="card-header"
    >
      <div
        class="font-semibold leading-none tracking-tight"
        data-testid="card-title"
      >
        Test Title
      </div>
      <div
        class="text-sm text-muted-foreground"
        data-testid="card-description"
      >
        Test Description
      </div>
    </div>
    <div
      class="p-6 pt-0"
      data-testid="card-content"
    >
      <p>
        Test content
      </p>
    </div>
    <div
      class="flex items-center p-6 pt-0"
      data-testid="card-footer"
    >
      <p>
        Test footer
      </p>
    </div>
  </div>
</div>
`;
